src/v/redpanda/admin/server.cc

```diff
@@ -2253,6 +2253,52 @@ admin_server::put_license_handler(std::unique_ptr<ss::http::request> req) {
     co_return ss::json::json_void();
 }
 
+ss::future<ss::json::json_return_type>
+admin_server::get_enterprise_handler(std::unique_ptr<ss::http::request>) {
+    if (!_controller->get_feature_table().local().is_active(
+          features::feature::license)) {
+        throw ss::httpd::bad_request_exception(
+          "Feature manager reports the cluster is not fully upgraded to "
+          "accept get enterprise requests");
+    }
+    using status = ss::httpd::features_json::enterprise_response::
+      enterprise_response_license_status;
+
+    const auto& license
+      = _controller->get_feature_table().local().get_license();
+    auto license_status = [&license]() {
+        auto present = license.has_value();
+        auto exp = present && license.value().is_expired();
+        if (exp) {
+            return status::expired;
+        }
+        if (present) {
+            return status::valid;
+        }
+        return status::not_present;
+    }();
+
+    auto& mgr = _controller->get_feature_manager();
+    const auto report = co_await mgr.invoke_on(
+      cluster::feature_manager::backend_shard,
+      [](const cluster::feature_manager& fm) {
+          return fm.report_enterprise_features();
+      });
+
+    ss::httpd::features_json::enterprise_response res;
+    res.license_status = license_status;
+    res.violation = license_status != status::valid && report.any();
+    for (std::size_t i = 0; i < report.size(); ++i) {
+        auto feat = static_cast<model::license_required_feature>(i);
```

Comment:
if the feature set were represented as a vector of enum then we could avoid this casting from integer to enum, right? is the size of the feature set important? it seems like right now it would be about 20 bytes max and one would only exist in memory for the lifetime of the request?

Reply:
>feature set were represented as a vector of enum

we want both enabled and disabled features, so I was choosing between `bitset` and `map<enum,bool>`, both of which are kinda smelly IMO. not really about size, just teasing out a "natural" representation for both the state of each feature and whether _any_ enterprise feature is ON. 

that said I think the answer is neither - 2x `vector<enum>` is 👌